test/martha/martha_v3.test.js

```diff
@@ -171,11 +176,16 @@ test.serial('martha_v3 calls the correct endpoints when only the accessUrl is re
         access_methods: [{ access_id: accessId, access_url: { url: gcsUrl } }]
     } = passportTestResponse;
     const passport = '"I am a passport"';
+    const clientKey = "client key";
+    const clientCert = "client cert";
+
     const drs = drsUrls(config.HOST_PASSPORT_TEST, objectId, accessId);
     const drsAccessUrlResponse = mockGcsAccessUrl(gcsUrl);
     getJsonFromApiStub.withArgs(drs.objectsUrl, null).resolves(passportTestResponse);
     getJsonFromApiStub.withArgs(ecmUrls('ras').passportUrl, terraAuth).resolves(passport);
-    postJsonToApiStub.withArgs(drs.accessUrl, null, {"passports": [passport]}).resolves(drsAccessUrlResponse);
+    postJsonToApiStub.withArgs(drs.accessUrl, null, {"passports": [passport]}, clientKey, clientCert).resolves(drsAccessUrlResponse);
+    accessSecretVersionStub.withArgs({name: config.rasClientMTLSKeySecretName}).resolves([{payload: {data: clientKey}}]);
+    accessSecretVersionStub.withArgs({name: config.rasClientMTLSCertSecretName}).resolves([{payload: {data: clientCert}}]);
```

Comment:
Should add this to the end of this test:
```
    sinon.assert.callCount(accessSecretVersionStub, 2);
```
Otherwise the test will pass even if these calls don't happen.

I had experimented with using the sinon API to declare these stub calls to be required, but it wasn't working for some reason that I couldn't figure out. The best compromise I came up with was to assert the number of expected calls at the end of the test function. It's not perfect because the test _could_ still pass if there are 2 calls but they don't match these calls. However, if the test is built so that the results of these stubs is necessary for the rest of the assertions, then call count should be good enough.

Reply:
I am inclined not to do this. Does the test care that these were called or does it only care that the post stubbed on the line above was called with the right key and cert? I think we care about the latter and the test as written will fail if the key and cert are not right.

Put another way, if we decide we need to cache the key and cert and these apis are not called the same way, do we want this test to fail? I don't think so.